src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -1570,14 +1635,17 @@ int decryptBlocks(GCM op, byte[] in, int inOfs, int inLen,
             }
 
             // Finish off the operation
-            if (inLen > TRIGGERLEN) {
-                int l = throttleData(op, in, inOfs, inLen, out, outOfs);
-                inOfs += l;
-                inLen -= l;
-                outOfs += l; // noop for ghash
-                len += l;
+            if (inLen >= PARALLEL_LEN) {
+                resultLen = GaloisCounterMode.implGCMCrypt(in, inOfs, inLen, in,
+                    inOfs, out, outOfs, gctr, ghash);
+                inOfs += resultLen;
+                inLen -= resultLen;
+                outOfs += resultLen;
+                len += resultLen;
             }
-            return len + op.doFinal(in, inOfs, inLen, out, outOfs);
+
+            ghash.doFinal(in, inOfs, inLen);
+            return len + gctr.doFinal(in, inOfs, inLen, out, outOfs);
```

Comment:
Why not just keep "return len + op.doFinal(in, inOfs, inLen, out, outOfs);"?

Reply:
same as above comment

Reply:
It probably got changed because I was cleaning up unused 'op' elsewhere